### PR TITLE
RangeSlider Component 

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

https://github.com/Royal-Navy/standards-toolkit/issues/220

## Overview

Implement the Range Slider component.

## Reason

>A range slider is an interactive component that allows the user to move a 'handle' to select a single value or range.

## Work carried out

- [x] Implement base functionality
- [x] Style component
- [x] Add additional stories
- [x] Allow user defined icons to be added to extremes of range
- [x] Display current value above handle on drag
- [x] Enable keyboard controls
- [x] Add automated tests
- [x] Create developer documentation
- [ ] Create design documentation

## Developer notes

~~TODO: Need to figure out logic for applying active state class to ticks when there are multiple handles.~~

### CODE REVIEW

**Review by hxltrhuxze — APPROVED** [review]

File: packages/react-component-library/src/components/RangeSlider/Slider.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  reversed,
+  disabled,
+  values,
+  onUpdate,
+  ...rest
+}) => {
+  const [sliderValues, setSliderValues] = useState(values)
+
+  const onUpdateHandler = (newValues: ReadonlyArray<number>): void => {
+    setSliderValues(newValues)
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/RangeSlider/Slider.tsx:44]: These two lines do not have test coverage

File: packages/react-component-library/src/components/RangeSlider/RangeSlider.test.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    it('should render a single handle', () => {
+      expect(wrapper.queryAllByTestId('rangeslider-handle').length).toBe(1)
+    })
+
+    it('should apply the correct classes', () => {
+      expect(
+        wrapper
+          .queryByTestId('rangeslider')
+          .classList.contains('rn-rangeslider')
+      ).toBe(true)
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/RangeSlider/RangeSlider.test.tsx]: There is also `toContains` which might provide a better error message if it fails.

File: packages/react-component-library/src/components/RangeSlider/RangeSlider.test.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+          step={step}
+          disabled
+        />
+      )
+    })
+
+    it('should apply the stateful `is-disabled` class', () => {
+      expect(
+        wrapper.queryByTestId('rangeslider').classList.contains('is-disabled')
+      ).toBe(true)
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/RangeSlider/RangeSlider.test.tsx]: See `toContains` comment

File: packages/react-component-library/src/components/RangeSlider/RangeSlider.test.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+          step={step}
+          reversed
+        />
+      )
+    })
+
+    it('should apply the stateful `is-reversed` class', () => {
+      expect(
+        wrapper.queryByTestId('rangeslider').classList.contains('is-reversed')
+      ).toBe(true)
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/RangeSlider/RangeSlider.test.tsx]: See `toContains` comment

File: packages/react-component-library/src/components/RangeSlider/Tick.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  domain: ReadonlyArray<number>
+  reversed?: boolean
+}
+
+function isActive(values: ReadonlyArray<number>, tickValue: number): boolean {
+  const valuesBelow = values.some(item => item <= tickValue)
+  const valuesAbove = values.some(item => item >= tickValue)
+
+  if (
+    (values.length === 1 && values[0] >= tickValue) ||
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/RangeSlider/Tick.tsx]: Wondering if some variable names and then a ternary operator could work well here.

**Review by thyhjwb6 — COMMENTED** [review]
Looks good 👍 

Comments are relatively minor. I also noticed that the `Veritcal` story is not working for me in Storybook.

File: packages/react-component-library/src/components/RangeSlider/Tick.tsx (hunk 5 above)
Comment (m7kvqbe1) [reply packages/react-component-library/src/components/RangeSlider/Tick.tsx]: Good shout - this has now been simplified

File: packages/react-component-library/src/components/RangeSlider/Slider.tsx (hunk 1 above)
Comment (m7kvqbe1) [reply packages/react-component-library/src/components/RangeSlider/Slider.tsx:44]: > Looks good 👍
> 
> Comments are relatively minor. I also noticed that the `Veritcal` story is not working for me in Storybook.

I'm going to implement the vertical story within another PR. I've removed it for the time being.

Comment (m7kvqbe1) [reply packages/react-component-library/src/components/RangeSlider/Slider.tsx:44]: I've added a test covering this now - also tests keyboard control as a result.

File: packages/react-component-library/src/components/RangeSlider/Tick.tsx [hunk 6]
```diff
@@ -14,15 +14,10 @@ function isActive(values: ReadonlyArray<number>, tickValue: number): boolean {
...
   const valuesAbove = values.some(item => item >= tickValue)
 
-  if (
-    (values.length === 1 && values[0] >= tickValue) ||
-    (valuesBelow && valuesAbove)
-  ) {
-    // below single handle || between multiple handles
-    return true
-  }
+  const belowSingleHandle = values.length === 1 && values[0] >= tickValue
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/RangeSlider/Tick.tsx:17]: Nice 👍

**Review by thyhjwb6 — APPROVED** [review]

**Pull request merged**